driver_test.go

```diff
@@ -661,3 +661,366 @@ func TestExecContextDml(t *testing.T) {
 	}
 
 }
+
+func TestExecContextDdl(t *testing.T) {
+
+	// Open db.
+	ctx := context.Background()
+	db, err := sql.Open("spanner", dsn)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer db.Close()
+
+	// Run tests.
+	tests := []struct {
+		name          string
+		given         string
+		withInsert    string
+		when          string
+		thenWantError bool
+		drop          string
+	}{
+		{
+			name: "create table ok",
+			when: `CREATE TABLE TestTable (
+				A   STRING(1024),
+				B  STRING(1024),
+			)	 PRIMARY KEY (A)`,
+			drop: `DROP TABLE TestTable`,
+		},
+		{
+			given: `CREATE TABLE TestTable (
+				A   STRING(1024),
+				B  STRING(1024),
+			)	 PRIMARY KEY (A)`,
+			name: "create table name duplicate",
+			when: `CREATE TABLE TestTable (
+				A   STRING(1024),
+				B  STRING(1024),
+			)	 PRIMARY KEY (A)`,
+			thenWantError: true,
+			drop:          `DROP TABLE TestTable`,
+		},
+		{
+			name: "create table syntax error",
+			when: `CREATE CREATE TABLE SyntaxError (
+				A   STRING(1024),
+				B  STRING(1024),
+				C   STRING(1024)
+			)	 PRIMARY KEY (A)`,
+			thenWantError: true,
+		},
+		{
+			name: "create table no primary key",
+			when: `CREATE TABLE NoPrimaryKey (
+				A   STRING(1024),
+				B  STRING(1024),
+			)`,
+			thenWantError: true,
+		},
+		{
+			name: "create table float primary key",
+			drop: "DROP TABLE FloatPrimaryKey",
```

Comment:
Please match the order across the test cases. The order in the struct definition is good as it matches the order they're used in the test which makes the test cases very readable

Reply:
matched